lib/routes/routeVeeam.js

```
@@ -67,8 +89,32 @@ function checkBucketAndKey(bucketName, objectKey, requestQueryParams, method, lo
         // Download relies on GETs calls with auth in query parameters, that can be
         // checked if 'X-Amz-Credential' is included.
         // Deletion requires that the tags of the object are returned.
-        if (requestQueryParams && Object.keys(requestQueryParams).length > 0
-            && !(method === 'GET' && (requestQueryParams['X-Amz-Credential'] || ('tagging' in requestQueryParams)))) {
+        const queryKeys = Object.keys(requestQueryParams || {});
+        const isPresignedGet = method === 'GET'
+            && (requestQueryParams
+                && (_hasQueryKeyCaseInsensitive(requestQueryParams, 'X-Amz-Credential')
+                    || _hasQueryKeyCaseInsensitive(requestQueryParams, 'tagging')));
+
+        if (!isPresignedGet && queryKeys.length > 0) {
```

With the previous logic, having !isPresignedGet && queryKeys.length > 0 was enough to return an error. 
Can we return an error directly here, and do the check unsupportedKeys right after, because here the 2 are mixed but I think it's not necessary ?